packages/lit-dev-content/site/docs/ssr/server-usage.md

```diff
@@ -8,6 +8,116 @@ eleventyNavigation:
+
+Allowing Promises in a sync iterable creates a kind of hybrid sync / async iteration protocol. Consumers of `RenderResult`s must check each value to see if it is a Promise or iterable and wait or recurse as needed.
+
+`@lit-labs/ssr` contains three utilities to do this for you:
+
+- `RenderResultReadable`
+- `collectResult()`
+- `collectResultSync()`
+
+#### `RenderResultReadable`
+
+`RenderResultReadable` - a Node `Readable` stream implementation that provides values from a `RenderResult`. This can be piped into a `Writable` stream, or passed to web server frameworks like Koa.
+
+This is the preferred way to handle SSR results when integrating with a streaming HTTP server or other stream-supprting API.
+
+```ts
+import {render} from '@lit-labs/ssr';
+import {RenderResultReadable} from '@lit-labs/ssr/lib/render-result-readable.js';
+
+// Using Koa to stream
+app.use(async (ctx) => {
+  const result = render(html`<my-element></my-element>`);
+  ctx.type = 'text/html';
+  ctx.body = new RenderResultReadable(result);
+});
+```
+
+#### `collectResult()`
+
+`collectResult(result: RenderResult): Promise<string>`
+
+`collectResult()` is an async function that takes a `RenderResult` and joins it into a string. It waits for Promises and recurses into nested iterables.
+
+##### Example
+```ts
+import {render} from '@lit-labs/ssr';
+import {collectResult} from '@lit-labs/ssr/lib/render-result.js';
+
+const result = render(html`<my-element></my-element>`);
+const html = await collectResult(result);
+```
+
+#### `collectResultSync()`
+
+`collectResultSync(result: RenderResult): Promise<string>`
+
+`collectResultSync()` is a sync function that takes a RenderResult and joins it into a string. It recurses into nested iterables, but _throws_ when it encounters a Promise.
+
+Because this function doesn't support async rendering, it's recommended to only use it when you can't await async functions.
+
+```ts
+import {render} from '@lit-labs/ssr';
+import {collectResultSync} from '@lit-labs/ssr/lib/render-result.js';
+
+const result = render(html`<my-element></my-element>`);
+// Throws if `result` contains a Promise!
+const html = collectResultSync(result);
+```
+
+### Render options
+
+`render()`'s second argument is a `RenderInfo` object that is used to pass options and current render state to components and sub-templates.
+
+The main options that can be set by callers are:
+
+* `deferHydration`: controls whether the top-level custom elements have a `defer-hydration` attribute added to signal that the elements should not automatically hydrate. This defaults to `false` so that top-level elements _do_ automatically hydrate.
+* `elementRenderers`: An array of `ElementRenderer` classes to use for rendering custom elements. By default this contains `LitElementRenderer` to render Lit elements. If can be set to include custom `ElementRenderer`s (documentation forthcoming), or set to an empty array to disable custom element rendering altogether.
```

Comment:
```suggestion
* `elementRenderers`: An array of `ElementRenderer` classes to use for rendering custom elements. By default this contains `LitElementRenderer` to render Lit elements. It can be set to include custom `ElementRenderer`s (documentation forthcoming), or set to an empty array to disable custom element rendering altogether.
```